.github/workflows/main.yml

```diff
@@ -0,0 +1,85 @@
+name: Nar
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build-macos:
+    name: Build on macos
+    runs-on: macos-11
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install build dependencies
+      run: |
+        brew update
+        brew install automake
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Build and test the package
+      run: mvn test
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: macos-build
+        path: target/nar/louis-*
+
+  build-windows:
+    name: Build for Windows
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+    - name: Install build dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install texinfo libc6-dev-i386 mingw-w64
+    - name: Build and test the package
+      run: mvn test -Pcross-compile -Dhost.os=w64-mingw32 -Dos.arch=x86_64
+    - name: Store the build arctifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: windows-build
+        path: target/nar/louis-*
+
+  nar:
+    name: Build for linux and package the nar
+    needs: [build-macos, build-windows]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-java@v2
+      with:
+        distribution: 'temurin'
+        java-version: '8'
+        cache: 'maven'
+        server-id: sonatype-nexus-snapshots # sonatype-nexus-staging
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+    - uses: actions/download-artifact@v3
+      with:
+        name: windows-build
+        path: target/nar
+    - uses: actions/download-artifact@v3
+      with:
+        name: macos-build
+        path: target/nar
+    - name: Build the nar
+      run: mvn --batch-mode test nar:nar-prepare-package nar:nar-package jar:jar
+    - name: Publish package
+      run: mvn --batch-mode deploy
```

Comment:
This shoud be `deploy:deploy` because `deploy` repeats some of the work that the previous command did.

Reply:
Ah, OK, I noticed that it was compiling again

Reply:
To make a release the artifacts should be signed (`pgp:sign`). Also it will be better to use nexus-staging-maven-plugin to do the deployment because that will automatically close the stage (one manual step less to do).

Reply:
Thanks

Reply:
So I added support for gpg signing. @bertfrees you need to update the values of the secrets `MAVEN_GPG_PASSPHRASE` and `MAVEN_GPG_PRIVATE_KEY`. The details on how to capture these are shown in the [advanced usage docs](https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#gpg) of the [setup-java](https://github.com/actions/setup-java) action.

Reply:
Ah yes, I already did :-)